ops/terraform/services/pipeline/modules/bfd_pipeline_manager/lambda_src/pipeline_manager.py

```diff
@@ -0,0 +1,374 @@
+    etl_bucket = s3_resource.Bucket(ETL_BUCKET_ID)
+    sqs_resource = boto3.resource("sqs", config=BOTO_CONFIG)
+    jenkins_job_runner_queue = sqs_resource.get_queue_by_name(QueueName=JENKINS_JOB_RUNNER_QUEUE)
+    ongoing_load_queue = sqs_resource.get_queue_by_name(QueueName=ONGOING_LOAD_QUEUE)
+except Exception as exc:
+    print(
+        f"Unrecoverable exception occurred when attempting to create boto3 clients/resources: {exc}"
+    )
+    sys.exit(0)
+
+
+class PipelineLoadType(str, Enum):
+    """Represents the possible types of data loads: either the data load is non-synthetic, meaning
+    that it contains production data and was placed within the root-level Incoming/Done folders of
+    the ETL bucket, or it is synthetic, meaning that it contains non-production, testing data and
+    was placed within the Incoming/Done folders within the Synthetic folder of the ETL bucket. The
+    value of each enum represents the name of the Incoming/Done folders' parent directory, with
+    empty string indicating that those paths have no parent"""
+
+    NON_SYNTHETIC = ""
+    SYNTHETIC = "Synthetic"
+
+
+class PipelineDataStatus(str, Enum):
+    """Represents the possible states of data: either data is available to load, or has been loaded
+    by the ETL pipeline. The value of each enum is the parent directory of the incoming file,
+    indicating status"""
+
+    INCOMING = "Incoming"
+    DONE = "Done"
+
+
+class RifFileType(str, Enum):
+    """Represents all of the possible RIF file types that can be loaded by the BFD ETL Pipeline. The
+    value of each enum is a specific substring that is used to match on each type of file"""
+
+    BENEFICIARY_HISTORY = "beneficiary_history"
+    BENEFICIARY = "bene"
+    CARRIER = "carrier"
+    DME = "dme"
+    HHA = "hha"
+    HOSPICE = "hospice"
+    INPATIENT = "inpatient"
+    OUTPATIENT = "outpatient"
+    PDE = "pde"
+    SNF = "snf"
+
+
+@dataclass
+class OngoingLoadQueueMessage:
+    """Represents a message in the ongoing load queue that acts as a sentinel indicating a
+    particular group is currently being loaded. This Lambda checks for the existence of these
+    messages in the queue, and will only start/stop the CCW pipeline instance depending on the
+    existence of such messages"""
+
+    load_type: PipelineLoadType
+    load_group: str
+    message_id: Optional[str] = None
+    receipt_handle: Optional[str] = None
+
+
+@dataclass
+class JenkinsJobRunnerQueueMessage:
+    """Represents the message that should be posted to the Jenkins job runner queue to invoke a
+    Jenkins job"""
+
+    job: str
+    parameters: "JenkinsTerraserviceJobParameters"
+
+
+@dataclass
+class JenkinsTerraserviceJobParameters:
+    """Represents the parameters for the BFD Pipeline Deploy Terraservice Jenkins Pipeline/job"""
+
+    env: str
+    create_ccw_pipeline_instance: bool
+
+
+def _check_ongoing_load_queue(timeout: int = 1) -> list[OngoingLoadQueueMessage]:
```

Comment:
Thoughts on re-naming this something like `_get_ongoing_load_queue_messages` ?

Reply:
Agreed, I think your suggested name is more specific and a better fit for what the function actually does.

I'll change it.